cmd/cortex/main.go

```diff
@@ -357,6 +354,16 @@ func setupIngester(
 }
 
 // setupRuler sets up a ruler.
-func setupRuler(chunkStore chunk.Store, cfg ruler.Config) (*ruler.Ruler, error) {
-	return ruler.New(chunkStore, cfg)
+func setupRuler(distributorCfg distributor.Config, chunkStore chunk.Store, cfg ruler.Config) (ruler.Worker, error) {
+	dist, err := distributor.New(distributorCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	externalURL, err := url.Parse(cfg.ExternalURL)
```

Comment:
`cfg.ExternalURL` is not wired up via a flag - should it be? Although the exact value doesn't seem to matter yet, as the comment above the field suggests.